spec/common/providers/identity.spec.ts

```diff
@@ -85,4 +90,857 @@ describe('identity', () => {
+            admin: true,
+            group_id: 'group123',
+          },
+          tenant_id: 'TENANT_ID',
+        },
+        oauth_id_token: 'ID_TOKEN',
+        oauth_access_token: 'ACCESS_TOKEN',
+        oauth_refresh_token: 'REFRESH_TOKEN',
+        oauth_token_secret: 'OAUTH_TOKEN_SECRET',
+        oauth_expires_in: 3600,
+        raw_user_info: JSON.stringify(rawUserInfo),
+      };
+      const context = {
+        locale: 'en',
+        ipAddress: '1.2.3.4',
+        userAgent: 'USER_AGENT',
+        eventId: 'EVENT_ID',
+        eventType:
+          'providers/cloud.auth/eventTypes/user.beforeCreate:oidc.provider',
+        authType: 'USER',
+        resource: {
+          service: 'identitytoolkit.googleapis.com',
+          name: 'projects/project-id/tenants/TENANT_ID',
+        },
+        timestamp: new Date(1000).toUTCString(),
+        additionalUserInfo: {
+          username: undefined,
+          providerId: 'oidc.provider',
+          profile: rawUserInfo,
+          isNewUser: true,
+        },
+        credential: {
+          claims: undefined,
+          accessToken: 'ACCESS_TOKEN',
+          expirationTime: new Date(time + 3600 * 1000).toUTCString(),
+          idToken: 'ID_TOKEN',
+          providerId: 'oidc.provider',
+          refreshToken: 'REFRESH_TOKEN',
+          secret: 'OAUTH_TOKEN_SECRET',
+          signInMethod: 'oidc.provider',
+        },
+        params: {},
+      };
+
+      expect(
+        identity.parseAuthEventContext(decodedJwt, 'project-id', time)
+      ).to.deep.equal(context);
+    });
+  });
+
+  describe('validateAuthResponse', () => {
+    it('should not throw on undefined request', () => {
+      expect(() => identity.validateAuthResponse('event', undefined)).to.not
+        .throw;
+    });
+
+    it('should throw an error if customClaims have a blocked claim', () => {
+      expect(() =>
+        identity.validateAuthResponse('beforeCreate', {
+          customClaims: { acr: 'something' },
+        })
+      ).to.throw(
+        'The customClaims claims "acr" are reserved and cannot be specified.'
+      );
+    });
+
+    it('should throw an error if customClaims size is too big', () => {
+      let str = '';
+      for (let i = 0; i < 1000; i++) {
+        str += 'x';
+      }
+
+      expect(() =>
+        identity.validateAuthResponse('beforeCreate', {
+          customClaims: { idk: str },
+        })
+      ).to.throw('The customClaims payload should not exceed 1000 characters.');
+    });
+
+    it('should throw an error if sessionClaims have a blocked claim', () => {
+      expect(() =>
+        identity.validateAuthResponse('beforeSignIn', {
+          sessionClaims: { acr: 'something' },
+        })
+      ).to.throw(
+        'The sessionClaims claims "acr" are reserved and cannot be specified.'
+      );
+    });
+
+    it('should throw an error if sessionClaims size is too big', () => {
+      let str = '';
+      for (let i = 0; i < 1000; i++) {
```

Comment:
same as above. You probably just want "a".repeat(1000)

Reply:
Fixed